### PR TITLE
[DEV-5546] New api/v2/disaster/award/amount/ endpoint

### PR DESCRIPTION
**Pull request by tony-sappe**
**Description:**
Provides the File C award outlay and obligation totals. Filterable by DEFC and award type code

**Technical details:**
Also cleaned up previously created tests in the new disaster app

**Requirements for PR merge:**

1. [x] Unit & integration tests updated
2. [x] API documentation updated
3. [x] Necessary PR reviewers:
    - [x] Backend
    - [x] Frontend
4. [x] Matview impact assessment completed (N/A)
5. [x] Frontend impact assessment completed
6. [x] Data validation completed
7. [x] Appropriate Operations ticket(s) created (N/A)
8. [x] Jira Ticket [DEV-5546](https://federal-spending-transparency.atlassian.net/browse/DEV-5546):
    - [x] Link to this Pull-Request
    - [x] Performance evaluation of affected API
    - [x] Before / After data comparison

**Area for explaining above N/A when needed:**
```
No edits to database required
```

### CODE REVIEW

File: usaspending_api/disaster/v2/views/award/amount.py [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+                        "submission_fiscal_month"
+                    ],
+                    submission__quarter_format_flag=False,
+                )
+            )
+
+        case_when_queries.append(
+            Q(
+                submission__reporting_fiscal_year=self.last_closed_quarterly_submission_dates["submission_fiscal_year"],
+                submission__reporting_fiscal_quarter=self.last_closed_quarterly_submission_dates[
```
Comment (kbard) [inline usaspending_api/disaster/v2/views/award/amount.py:79]: Is this (and monthly above) correct?  Should we be summing `transaction_obligated_amount` for the year up to and including last closed period?  And should `gross_outlay_amount_by_award_cpe` be following that crazy "sum all previous final of fy values" logic?

Comment (tony-sappe) [reply usaspending_api/disaster/v2/views/award/amount.py:79]: Not sure. This is copy-paste from other code which was (hopefully) vetted and should basically be the ORM you and I worked on last week

**Comment by tony-sappe** [comment]
> `TypeError: 'NoneType' object is not iterable` if `award_type_codes` is not provided in the filter because it's not required in `DisasterBase`. Do we want to require it for this endpoint or do we want to not attempt to filter on it if it's not provided?

@kbard good catch. I mistakenly thought `self.required_filters` made the filter keys required, when at second glance the list is simply assembling the tinyshield models to validate the request. Will update

File: usaspending_api/disaster/v2/views/disaster_base.py [hunk 2]
```diff
@@ -26,38 +26,37 @@ def covid_def_code_strings():
 
 def latest_gtas_of_each_year_queryset():
     q = Q()
-    for final_for_fy in finals_for_fy(False):
-        q |= Q(fiscal_year=final_for_fy[0]) & Q(fiscal_period=final_for_fy[1])
-    for final_for_fy in finals_for_fy(True):
-        q |= Q(fiscal_year=final_for_fy[0]) & Q(fiscal_period=final_for_fy[1])
+    for final_for_fy in finals_for_all_fy():
+        q |= Q(fiscal_year=final_for_fy[0]) & Q(fiscal_period=final_for_fy[2])
```
Comment (kbard) [inline usaspending_api/disaster/v2/views/disaster_base.py:30]: Up to you if you want to "fix" this, but GTAS only has monthly data so there's no sense querying quarters.  Didn't notice that before.  A simple `if not final_for_fy[1]:` before the `q |=` should eliminate them.

Comment (tony-sappe) [reply usaspending_api/disaster/v2/views/disaster_base.py:30]: Wrong PR, look at #2567 

I thought about quarters not being separate from monthly in GTAS, idea makes sense

File: usaspending_api/disaster/v2/views/award/amount.py (hunk 1 above)
Comment (tony-sappe) [reply usaspending_api/disaster/v2/views/award/amount.py]: Fixed code in another PR

File: usaspending_api/disaster/v2/views/disaster_base.py (hunk 2 above)
Comment (tony-sappe) [reply usaspending_api/disaster/v2/views/disaster_base.py]: Done

**Pull request merged**